maintainer/channels.py

```diff
@@ -0,0 +1,145 @@
+            channels_basic_info.append(channel)
+        return channels_basic_info
+
+    def channels_detailed_information(self):
+        '''
+        Method is returning more data than channels_basic_info method.
+        In this method we are receiving important data such as: channel_topic
+        or seconds_empty that allow us to see if there is any status written
+        into topic or if channel is used at all.
+        There is no input value from outside of class.
+        Output: channels_detailed_info as dictionary of dictionaries in form of:
+            cid: detailed_info of one channel
+        '''
+        channels_basic_info = self.channels_basic_info()
+        channels_detailed_info = {}
+        for channel in channels_basic_info:
+            channels_detailed_info[channel["cid"]] = self._requests_get(
+                'channels/{}'.format(channel["cid"]))
+        return channels_detailed_info
+
+    def channels_to_quarantine(self):
+        '''
+        Method is using channels_detailed_information for getting seconds_empty
+        and channel_topic values in order to specify if there should be put
+        status of quarantine on channels that meet certain condition.
+        There is no input taken from outside of class.
+        Output: Method is returning short information that it found channels
+        that needed attention.
+        '''
+        channels_detailed_info = self.channels_detailed_information()
+        for channel_id in channels_detailed_info:
+            channel = channels_detailed_info[channel_id]
+            if int(channel["seconds_empty"]) > self.qtime:
+                payload = {'channel_topic': 'quarantine'}
+                self._requests_post('channels/{}/topic'.format(channel_id), payload)
+        return "quarantined {}".format(payload)
+
+    def delete_parents(self):
+        '''
+        Method is used to delete channels that are parents to other channels.
+        We are using this method after delete_children because it deletes
+        children of parent as well.
+        There is no input taken from outside of class.
+        Output: short information which parent channels were deleted.
+        '''
+        channels_data = self.channels_detailed_information()
+        for channel_id in channels_data:
+            print(channel_id)
+            channel = channels_data[channel_id]
+            if channel["channel_topic"] != 'protected' and channel[
+                    "channel_topic"] == 'quarantine' and int(
+                        channel["seconds_empty"]) > self.qtime:
+                            self._requests_delete('channels/{}'.format(channel_id))
+            return "deleted channels {}".format(channel_id)
+
+    def delete_children(self):
+        '''
+        Method is used for deleting channels that are having children status.
+        There is no input taken from outside of class.
+        Output we are receiving list of children channels deleted by method.
```

Comment:
Wrong! Look at the return statement! Misguiding

Reply:
Fixed